commit 8604e532e35bd5c1e9d120dc53eb2f77aac971d4 (refs/changes/46/1246/1)
Author: Pedro Palazon Candel <pedro@joyent.com>
Date:   2017-01-13T10:57:51+01:00 (2 years, 9 months ago)
    
    'sdcadm experimental update SOME-AGENT' fails if there are no instances yet

diff --git a/lib/procedures/update-agent-v1.js b/lib/procedures/update-agent-v1.js
index 83aa868..a547353 100644
--- a/lib/procedures/update-agent-v1.js
+++ b/lib/procedures/update-agent-v1.js
@@ -177,6 +177,11 @@ UpdateAgentV1.prototype.execute = function uaExecute(opts, callback) {
             },
 
             function checkCNsAvailability(_, next) {
+                // Move into the next step if we have zero instances:
+                if (change.insts.length === 0) {
+                    next();
+                    return;
+                }
                 sdcadm.cnapi.listServers({
                     uuids: change.insts.map(function (i) {
                         return (i.server);
